driver/src/contracts/stablex_contract.rs

```diff
@@ -108,20 +114,45 @@ impl StableXContract for StableXContractImpl {
         Ok(Duration::from_secs(remaining_seconds.as_u64()))
     }
 
+    fn get_last_block_for_batch(&self, batch_id: u32) -> Result<Option<u64>> {
+        const BATCH_DURATION: u64 = 300;
+
+        let web3 = self.instance.raw_instance().web3();
+        let get_block = |block_number: BlockNumber| -> Result<_> {
+            web3.eth()
+                .block(block_number.into())
+                .wait()?
+                .ok_or_else(|| anyhow!("block {:?} is missing", block_number))
+        };
+
+        let mut current_block = get_block(BlockNumber::Latest)?;
```

Comment:
Maybe we should start with pending here?

Reply:
I tried this and ran into some issues. So, from what I understood we want the entire orderbook to be retrieved at a specific block height. That means that we need an actual block number to retreive the orderbook at. If we use pending, it is possible that a solution gets submitted partway through the orderbook and modifies orders and balances that have not yet been queried and then get included once we reach that page (the current problem we have). Also, querying at block `latest + 1` doesn't seem to work.

Reply:
Yeah, that was an oversight. If we don't use "pending" here, this method might return `None` when that last block in the previous batch is indeed "latest".

Reply:
I'm curious about the reason we switched back to latest?

Reply:
There is an issue with web3-rs `getBlock` when specifying `BlockNumber::Pending`. 

>  Apr 22 18:28:21.080 ERRO [driver::orderbook::shadow_orderbook] encountered an error reading the orderbook with the shadow reader: Decoder error: Error("invalid type: null, expected a 0x-prefixed hex string with length of 512", line: 0, column: 0)

I will investigate more and file an issue in their repo. For now, I think Latest should work.